web/ui/src/features/component/TableHead.tsx

```diff
@@ -0,0 +1,39 @@
+import { useState } from 'react';
+
+interface Props {
+  headers: string[];
+  handleSorting?: (header: string, sortOrder: string) => void;
+}
+
+const TableHead = ({ headers, handleSorting }: Props) => {
+  const [sortField, setSortField] = useState('');
+  const [order, setOrder] = useState('asc');
+
+  const handleSortingChange = (header: string) => {
+    const sortOrder = header === sortField && order === 'asc' ? 'desc' : 'asc';
+    setSortField(header);
+    setOrder(sortOrder);
+    if (handleSorting !== undefined) {
```

Comment:
the setters should be inside of the if statement because if handleSorting is undefined then there is not point of changing the value of sortField and order.